### ISSUE TITLE
Why define type="key" and type="query" in mask() in modules.py?

### ISSUE THREAD

**Issue by moonlight1776** [Kyubyong/transformer#82]
Hi, Kyubyong

I am confused the  **mask** form!
I find a concept called **padding mask**, but, after the projection of Dense, 
`Q = tf.layers.dense(queries, d_model)
K = tf.layers.dense(keys, d_model) # (N, T_k, d_model)
V = tf.layers.dense(values, d_model) # (N, T_k, d_model)` 
the 0 vector of padding will be calculated by `wx + b`,and will equal [b1, b2...bn], n=d_model.
`outputs = tf.matmul(Q, tf.transpose(K, [0, 2, 1]))`
I think there will never be a 0 vecter in outputs.
So, I am confused about the "type=keys"and "type=values".
`def mask(inputs, queries=None, keys=None, type=None):
if type in ("k", "key", "keys"):
...
elif type in ("q", "query", "queries"):
...`

**Comment by superhg2012** [comment]
@moonlight1776  have you figured out this issue? I am also confused !
